subsys/net/lib/rest_client/src/rest_client.c

```diff
@@ -221,6 +238,18 @@ static int rest_client_sckt_connect(int *const fd,
 		goto clean_up;
 	}
 
+	if (*timeout_ms != SYS_FOREVER_MS) {
+		/* Take time used for socket connect into account */
+		sckt_connect_time = k_uptime_get() - sckt_connect_start_time;
+
+		/* Check if timeout has already elapsed */
+		if (sckt_connect_time >= *timeout_ms) {
+			LOG_WRN("Timeout occurred during socket connect");
+			return -ETIMEDOUT;
+		}
+		*timeout_ms -= sckt_connect_time;
+	}
```

Comment:
This was my initial idea that I'll move timeout handling for socket connect into this function. However, I started to rethink that maybe this would be better and easier to understand in its original place in `rest_client_do_api_call()`.

Reply:
Do you mean that timeout during DNS query would be handled in `rest_client_sckt_connect()` and during socket connect in `rest_client_do_api_call()`? To me the current implementation looks fine.

Reply:
Yep, leaving socket connect timeout handling into `rest_client_do_api_call()` would avoid having `timeout_ms` as in/out parameter for this function. I'm fine either way.